nextcord/polls.py

```diff
@@ -0,0 +1,370 @@
+
+
+class PollCreateRequest:
+    __slots__ = (
+        "question",
+        "answers",
+        "duration",
+        "allow_multiselect",
+        "layout_type",
+    )
+    """
+    A poll create request.
+    You must use this to create a poll.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    question: :class:`PollMedia`
+        The question of the poll. Current only `text` is supported.
+
+    """
+
+    def __init__(
+        self,
+        question: PollMedia,
+        answers: List[PollAnswer],
+        duration: int,
+        allow_multiselect: bool,
+        layout_type: Optional[PollLayoutType] = MISSING,
+    ) -> None:
+        self.question: PollMedia = question
+        self.answers: List[PollAnswer] = answers
+        self.duration: int = duration
+        self.allow_multiselect: bool = allow_multiselect
+        self.layout_type: Optional[PollLayoutType] = layout_type
+
+    def to_dict(self) -> PollCreateRequestPayload:
+        payload: PollCreateRequestPayload = {
+            "question": self.question.to_dict(),
+            "answers": [answer.to_dict() for answer in self.answers],
+            "duration": self.duration,
+            "allow_multiselect": self.allow_multiselect,
+        }
+        if self.layout_type:
+            payload["layout_type"] = self.layout_type.value
+        return payload
+
+    def add_answer(self, text: str, emoji: Optional[EmojiInputType] = None) -> Self:
+        """Add a choice i.e. answer to the poll.
+        This function returns the class instance to allow for fluent-style chaining.
+
+        Parameters
+        ----------
+        text: :class:`str`
+            The text description of the choice.
+        emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`, :class:`str`]]
+            The emoji description of the choice.
+
+        Returns
+        -------
+        The poll create request object.
+        """
+        self.answers.append(PollAnswer(poll_media=PollMedia(text=text, emoji=emoji)))
+        return self
+
+
+class Poll:
+    """A poll object.
+
+
+    Attributes
+    ----------
+    message: :class:`Message`
+        The message this poll belong to.
+    question: :class:`PollMedia`
+        The question being asked in a poll.
+    answers: List[:class:`PollAnswer`]
+        The answers of the poll.
+    expiry: :class:`datetime.datetime`
+        The expiry of the poll.
+    allow_multiselect: :class:`bool`
+        Whether this poll allow people to select more than 1 option.
+    layout_type: :class:`PollLayoutType`
+        The layout type of the poll.
+    results: Optional[:class:`PollResults`]
+        The results of the poll.
+    """
+
+    __slots__ = (
+        "message",
+        "question",
+        "_state",
+        "answers",
+        "expiry",
+        "allow_multiselect",
+        "layout_type",
+        "results",
+    )
+
+    def __eq__(self, other) -> bool:
+        return self.message.id == other.message.id
```

Comment:
I suggest to compare it like this: https://github.com/MaskDuck/nextcord/blob/01f1850d34fab582f3bcc1b8602a172930da417b/nextcord/mixins.py#L23-L29
Additionally, you can add `__ne__`.